src/metadata_encoding.rs

```diff
@@ -0,0 +1,203 @@
+use bytes::Bytes;
+use http::header::HeaderValue;
+use std::fmt;
+use std::error::Error;
+use std::hash::Hash;
+
+/// A possible error when converting a `MetadataValue` from a string or byte
+/// slice.
+#[derive(Debug, Default)]
```

Comment:
I'd also remove the `Default` from this error.

Reply:
Done.